pghrep/src/pluginsutil.go

```diff
@@ -0,0 +1,151 @@
+package main
+
+import (
+	"fmt"
+    "io/ioutil"
+    "log"
+    "math/rand"
+    "os"
+    "os/exec"
+    "path/filepath"
+    "plugin"
+)
+
+
+type Preparer interface {
+	Prepare(map[string]interface{}) map[string]interface{}
+}
+
+// loader keeps the context needed to find where plugins and
+// objects are stored.
+type loader struct {
+    pluginsDir string
+    objectsDir string
+    binDir string
+}
+
+func newLoader() (*loader, error) {
+    // The directory that will be watched for new plugins.
+    wd, err := os.Getwd()
+    if err != nil {
+        return nil, fmt.Errorf("could not find current directory: %v", err)
+    }
+    pluginsDir := filepath.Join(wd, "plugins")
+    binDir := filepath.Join(wd, "bin")
+    //tmp := filepath.Join(wd, "bin")
+
+    // The directory where all .so files will be stored.
+    tmp, err := ioutil.TempDir("", "")
+    if err != nil {
+        return nil, fmt.Errorf("could not create objects dir: %v", err)
+    }
+    return &loader{pluginsDir: pluginsDir, objectsDir: tmp, binDir: binDir}, nil
+}
+
+func (l *loader) destroy() { 
+    //os.RemoveAll(l.objectsDir)
+}
+
+func (l *loader) compileAndRun(name string, data map[string]interface{}) (map[string]interface{}, error) {
+    obj, err := l.compile(name)
+    if err != nil {
+        return nil, fmt.Errorf("could not compile %s: %v", name, err)
+    }
+    defer os.Remove(obj)
+
+    result, err := l.call(obj, data)
+    if err != nil {
+        return nil, fmt.Errorf("could not run %s: %v", obj, err)
```

Comment:
`Cannot run`

(or, if you want, let's use `Could...`, bug starting with capital `C`)